src/transformers/trainer.py

```diff
@@ -951,6 +951,17 @@ def create_optimizer_and_scheduler(self, num_training_steps: int):
             optimizer = self.optimizer
         self.create_scheduler(num_training_steps=num_training_steps, optimizer=optimizer)
 
+    def get_decay_parameter_names(self, model) -> List[str]:
+        """
+        Get all parameter names that weight decay will be applied to
+
+        Note that some models implement their own layernorm instead of calling nn.LayerNorm, weight decay could still
+        apply to those modules since this function only filter out instance of nn.LayerNorm
```

Comment:
this could be easily overlooked and i am not sure what the best solution is. i could see that we (1) add all variants of layernorm - which introduce one more thing to maintain for future models; or (2) log it to make it more visible - which might again be overlooked; or (3) just document this in the docstring

Reply:
I think this is okay, the `ALL_LAYERNORM_LAYERS` should contain all the custom layer norms (from `transformers` modeling files) and should be updated if that is not the case